pyproject.toml

```diff
@@ -71,33 +71,35 @@ all = [
     "astropy[ipython]",
     "astropy[jupyter]",
     # Install all remaining optional dependencies
-    "certifi",
-    "dask[array]>=2022.8.1",
-    "h5py",
-    "pyarrow>=10.0.1",
-    "beautifulsoup4",
-    "html5lib",
-    "bleach",
-    "pandas>=2.0",
-    "sortedcontainers",
-    "pytz",
-    "jplephem",
-    "mpmath",
-    "asdf-astropy>=0.3",
-    "bottleneck",
+    "certifi>=2022.6.15.1",
+    "dask[array]>=2022.5.1",
+    "h5py>=3.8.0", # 2023-01-23
+    "pyarrow>=10.0.1", # 2022-11-22
+    "beautifulsoup4>=4.9.3", # 2020-10-03, imposed by pandas==2.0
+    "html5lib>=1.1", # 2020-06-23
+    "bleach>=3.2.1", # 2020-09-18
+    "pandas>=2.0", # 2023-04-03
+    "sortedcontainers>=1.5.7", # 2016-12-23 (older versions may work)
+    "pytz>=2016.10", # (older versions may work)
+    "jplephem>=2.6", # 2016-12-20 (older versions may work)
+    "mpmath>=1.2.1", # 2021-02-09
+    "asdf>=2.8.3",  # via asdf-astropy==0.3.*
+    "asdf-astropy>=0.3", # 2022-11-29
+    "bottleneck>=1.3.3", # 2022-02-22
     "fsspec[http]>=2023.4.0",
     "s3fs>=2023.4.0",
 ]
 # The base set of test-time dependencies.
 test = [
-    "coverage",
-    "pre-commit",
-    "pytest>=7.0",
-    "pytest-doctestplus>=0.12",
-    "pytest-astropy-header>=0.2.1",
-    "pytest-astropy>=0.10",
-    "pytest-xdist",
-    "threadpoolctl",
+    "coverage>=6.4.4", # 2022-08-16 (first version with cp311 wheels)
+    # 2020-12-08 (arbitrary), lower bound is not tested beyond installation
+    "pre-commit>=2.9.3",
+    "pytest>=7.3.0", # 2023-04-08
+    "pytest-doctestplus>=0.12", # 2022-02-25
+    "pytest-astropy-header>=0.2.1", # 2022-03-10
+    "pytest-astropy>=0.10.0", # 2022-04-12
```

Comment:
I wonder if some of the `pytest-` pins are outdated. Most of them should have come from `pytest-astropy` meta package.

Reply:
this is actually more restrictive than what's specified in `pytest-astropy` 0.10.0.
Furthermore, `--resolution=lowest-direct` doesn't take the lower bound of *indirect* dependencies: we should still specify some lower bound for ourselves if we really mean to check compatibility with anything older than the most recent version of these packages.

https://github.com/astropy/pytest-astropy/blob/v0.10.0/setup.cfg

Reply:
Well, uv also supports `--resolution=lowest`, which is probably closer to what you expect here, but it's much less likely to work for us given how large our fully resolved dependency tree, so, I'm going for the strategy with the highest chance of success first. We can maybe explore this option later on if you'd like, but I must say I am pessimistic about its odds.

Reply:
See https://github.com/astropy/astropy/issues/16950#issuecomment-2334840984